pkg/instrumentors/bpffs/bpfsfs.go

```diff
@@ -14,5 +14,54 @@
 
 package bpffs
 
+import (
+	"fmt"
+	"os"
+
+	"go.opentelemetry.io/auto/pkg/log"
+
+	"golang.org/x/sys/unix"
+
+	"go.opentelemetry.io/auto/pkg/process"
+)
+
 // BPFFsPath is the system path to the BPF file-system.
-const BPFFsPath = "/sys/fs/bpf"
+const bpfFsPath = "/sys/fs/bpf"
+
+// PathForTargetApplication returns the path to the BPF file-system for the given target.
+func PathForTargetApplication(target *process.TargetDetails) string {
+	return fmt.Sprintf("%s/%d", bpfFsPath, target.PID)
+}
+
+// Mount mounts the BPF file-system for the given target.
+func Mount(target *process.TargetDetails) error {
+	if !isBPFFSMounted() {
+		// Directory does not exist, create it and mount
+		if err := os.MkdirAll(bpfFsPath, 0755); err != nil {
+			return err
+		}
+
+		err := unix.Mount(bpfFsPath, bpfFsPath, "bpf", 0, "")
+		if err != nil {
+			return err
+		}
+	}
+
+	return os.Mkdir(PathForTargetApplication(target), 0755)
+}
+
+func isBPFFSMounted() bool {
+	var stat unix.Statfs_t
+	err := unix.Statfs(bpfFsPath, &stat)
+	if err != nil {
+		log.Logger.Error(err, "failed to statfs bpf filesystem")
+		return false
+	}
+
+	return stat.Type == unix.BPF_FS_MAGIC
+}
+
+// Cleanup removes the BPF file-system for the given target.
+func Cleanup(target *process.TargetDetails) error {
```

Comment:
Do we need to call this explicit ?

Reply:
Yes, when detecting that a process is dead and unloading the eBPF programs, this should also be invoked to cleanup the relevant data structures in BPFFS